app/src/ui/history/sidebar.tsx

```diff
@@ -53,9 +53,9 @@ export class HistorySidebar extends React.Component<IHistorySidebarProps, {}> {
   public render() {
     return (
       <CommitList
-        repository={this.props.repository}
-        commits={this.props.commits}
-        history={this.props.history.history}
+        gitHubRepository={this.props.repository.gitHubRepository}
+        commitLookup={this.props.commitLookup}
+        commits={this.props.history.history}
```

Comment:
Whats the difference between `commits` and `localCommitSHAs` in this file? I noticed that `commits` is just an enumeration of `commitLookup` which is keyed by commit SHA.

Reply:
`localCommitSHAs` are how we track unpublished commits for the current branch (for example, whether to display the Undo Commit hint in the commit form) and they should be the first commits in `commits` (if any exist)